### PR TITLE
Fix dependently-typed object functions crashes at runtime

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/25196

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/VariableReturnType.java [hunk 1]
```diff
@@ -101,6 +102,14 @@ public static Object getValue(BTypedesc td) {
         return getValue(td.getDescribingType());
     }
 
+    public static Object getObjectValue(ObjectValue objectValue, BTypedesc td) {
+        BType describingType = td.getDescribingType();
+        if (describingType.getTag() == STRING_TAG) {
+            objectValue.set(new BmpStringValue("fname"), getValue(describingType));
```
Comment (Kishanthan) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/VariableReturnType.java:108]: Should we have to return after this line?

**Comment by pubudu91** [comment]
Shouldn't I be able to do something like the following? 
```ballerina
type Person object {
    string fname;
    string lname;

    function init(string fname, string lname) {
        self.lname = lname;
        self.fname = fname;
    }

    function getFullName(typedesc<anydata> td)  returns td = @java:Method {
        class: "xyz.pubudu.Hello",
        name: "getObject",
        paramTypes: ["org.ballerinalang.jvm.values.api.BTypedesc"]
    } external;
};
```
The above results in the following error:
```
error: pubudu/fizz:0.1.0::main.bal:30:5: {ballerina/java}METHOD_NOT_FOUND 'No such public method 'getObject' that matches with parameter types '(org.ballerinalang.jvm.values.api.BTypedesc)' found in class 'class xyz.pubudu.Hello''
```
In my native class, I have the following method:
```java
public static BString getObject(ObjectValue objectValue, BTypedesc td) {
        BType describingType = td.getDescribingType();
        if (describingType.getTag() == STRING_TAG) {
            return objectValue.getStringValue(new BmpStringValue("fname")).concat(new BmpStringValue(" ")).concat(
                    objectValue.getStringValue(new BmpStringValue("lname")));
        }
        return new BmpStringValue("Invalid");
}
```

**Comment by warunalakshitha** [comment]
@pubudu91 Your example is working when we add  paramTypes: ["org.ballerinalang.jvm.values.ObjectValue", "org.ballerinalang.jvm.values.api.BTypedesc"]. Is that the expected behavior

**Comment by warunalakshitha** [comment]
I have fix the issue with @pubudu91 example and add relevant testcases as well.

**Pull request merged**